pysb/export/json.py

```diff
@@ -0,0 +1,228 @@
+"""
+Module containing a class for exporting a PySB model to JSON
+
+For information on how to use the model exporters, see the documentation
+for :py:mod:`pysb.export`.
+"""
+
+from __future__ import absolute_import
+from pysb.export import Exporter
+from pysb.bng import generate_equations
+import json
+from pysb.core import Model, MultiState, KeywordMeta, Parameter, Expression
+
+
+class JsonExporter(Exporter):
+    """A class for returning the JSON for a given PySB model.
+
+    Inherits from :py:class:`pysb.export.Exporter`, which implements
+    basic functionality for all exporters.
+    """
+
+    def export(self, include_netgen=False):
+        """Generate the corresponding JSON for the PySB model associated
+        with the exporter.
+
+        Parameters
+        ----------
+        include_netgen: bool
+            Include cached network generation data (reactions, species,
+            local function-derived parameters and expressions) if True.
+
+        Returns
+        -------
+        string
+            The JSON output for the model.
+        """
+        return json.dumps(self.model, cls=PySBJSONWithNetworkEncoder
+                          if include_netgen else PySBJSONEncoder)
+
+
+class PySBJSONEncoder(json.JSONEncoder):
+    FORMAT = 1
```

Comment:
I've switched to `protocol`, rather than `format`.

I've added some documentation to cover the basics of the format. The protocol number should get bumped when new features are added which affect model simulation, or prevent it being loaded by the decoder. I don't think it's a bad thing if this happens, so long as the decoder maintains backwards compatibility.